java/ql/lib/semmle/code/java/security/XmlParsers.qll

```diff
@@ -628,15 +554,22 @@ class XmlReaderConfig extends ParserConfig {
   }
 }
 
-private module ExplicitlySafeXmlReaderFlowConfig implements DataFlow::ConfigSig {
+deprecated private module ExplicitlySafeXmlReaderFlowConfig implements DataFlow::ConfigSig {
```

Comment:
What's the reason for deprecating these private modules instead of simply using the new `SimpleGlobal` flows elsewhere?

Reply:
That's because they are transitively used in a non-private member predicate that I also deprecated. I then also marked the private declarations that we eventually want to remove as deprecated as well in order to make it clear that they shouldn't be accidentally used for anything else.

Reply:
I see. Wouldn't it be simpler to use the new flows in the non-private member predicate instead?

Reply:
> Wouldn't it be simpler to use the new flows in the non-private member predicate instead?

That's certainly an alternative, but the point is that with the new flows we're only calculating sink-reachability and not source-sink pairs, so the couple of places where we accidentally exposed a public member predicate that links the sink to the source, then we need an additional computation. So instead of freshly introducing this via the light-weight api only to use it in a deprecated predicate, then I thought it was simpler to leave the old implementation.

I don't think it matters that much though, as it's all inside a now-deprecated and very likely completely unused predicate.

Reply:
👍 Makes sense, I just mentioned it because it seemed that we could remove even more code if we did it that way, but as you said it doesn't matter much indeed. Let's leave it as it is.